lib/super_digit.rb

```diff
@@ -1,15 +1,29 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - n?
+# Space Complexity - n? since methods are chained?
+  # n.to_s.split('').map{ |num| num.to_i }.sum
+  # n t:1 s:1
+  # .to_s t:1 s:n
+  # .split('') t:n s:n
+  # .map t:n s:n
+  # .sum t:n s:1
+
 def super_digit(n)
-   
+  if n.to_s.split('').length < 2
+    return n
+  end
+  return super_digit(n.to_s.split('').map{ |num| num.to_i }.sum)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - n
+# Space Complexity - n*m - because of k
 def refined_super_digit(n, k)
```

Comment:
```suggestion
# Time Complexity - (nk)^2
# Space Complexity - n*k - because of k
def refined_super_digit(n, k)
```

This works, but it's not quite the most efficient way, because you can make it more efficient using dynamic programming, see below.